### PR TITLE
nextcloud-client: 3.1.3 -> 3.2.0

### PR DESCRIPTION
**Pull request by Kranzes**
###### Motivation for this change
Bump to the latest version. (at the time of making this PR).

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/nextcloud-client/default.nix [hunk 1]
```diff
@@ -20,13 +20,13 @@
...
   pname = "nextcloud-client";
-  version = "3.1.3";
+  version = "3.2.0";
 
   src = fetchFromGitHub {
     owner = "nextcloud";
     repo = "desktop";
     rev = "v${version}";
-    sha256 = "sha256-8Ql6tOvWOjAvMJA87WlT9TbpnbciBsjDxRuYlMVi/m8=";
+    sha256 = "1nklsa2lx9ayjp8rk1mycjysqqmnq47djig0wygzna5mycl3ji06";
```
Comment (caugner) [inline pkgs/applications/networking/nextcloud-client/default.nix:29]: I think we should keep the `sha256-` prefix.

```suggestion
    sha256 = "sha256-1nklsa2lx9ayjp8rk1mycjysqqmnq47djig0wygzna5mycl3ji06";
```

**Review by caugner — CHANGES_REQUESTED** [review]
LGTM except for the missing hash prefix

File: pkgs/applications/networking/nextcloud-client/default.nix (hunk 1 above)
Comment (Kranzes) [reply pkgs/applications/networking/nextcloud-client/default.nix:29]: 1. How do you obtain a hash with that prefix?
2. I do not see that hash prefix being used much.
3. Why should you use one over the other?

Comment (caugner) [reply pkgs/applications/networking/nextcloud-client/default.nix:29]: I thought I had read somewhere that `sha256` is going to be replaced by `hash` at some point, which would make it necessary to specify the hash algorithm inside the hash string. However, I don't find that post anymore, so I'm really not sure.

@bryanasdev000 Do you remember why you added the `sha256-` prefix in 4515dfb6a23192d19b0eedd8564c888ce2474cfc?

Comment (caugner) [reply pkgs/applications/networking/nextcloud-client/default.nix:29]: Ah, you have already answered this question here: https://github.com/NixOS/nixpkgs/pull/113731/files#r580222460

> SRI hashes (https://www.w3.org/TR/SRI/) combine the hash algorithm and  a base-64 hash. This allows more concise and standard hash specifications.

https://github.com/NixOS/nix/commit/6024dc1d97212130c19d3ff5ce6b1d102837eee6

So I guess we could write:

```suggestion
    hash = "sha256-1nklsa2lx9ayjp8rk1mycjysqqmnq47djig0wygzna5mycl3ji06";
```

Comment (caugner) [reply pkgs/applications/networking/nextcloud-client/default.nix:29]: Looks like it's just a personal preference, not anything that has been decided yet, so I'll update my review accordingly.

**Review by caugner — APPROVED** [review]
LGTM

File: pkgs/applications/networking/nextcloud-client/default.nix (hunk 1 above)
Comment (caugner) [reply pkgs/applications/networking/nextcloud-client/default.nix:29]: Yeah, sorry about the confusion. The mention on that PR tricked my memory.

**Comment by ethancedwards8** [comment]
LGMT too.

**Review by ethancedwards8 — APPROVED** [review]

File: pkgs/applications/networking/nextcloud-client/default.nix (hunk 1 above)
Comment (bryanasdev000) [reply pkgs/applications/networking/nextcloud-client/default.nix:29]: Just a note, if in any case the version is changed by a bot like ryantm's, the hash returns to the "new" model by default. In the end it doesn't make much difference but it is good to keep in mind.
Another thing, I use unstable Nix so by default it gives me this hash when I update git and I try to build it (my default way to know new hashes lol) I'm not sure if nix-prefetch- * is still using the old model.

Comment (Kranzes) [reply pkgs/applications/networking/nextcloud-client/default.nix:29]: I am using stable Nix and it gives me the non-SRI hashes.

**Pull request merged**